src/bindings/python/src/pyopenvino/graph/shape.cpp

```diff
@@ -40,10 +40,24 @@ void regclass_graph_Shape(py::module m) {
     shape.def("__setitem__", [](ov::Shape& self, size_t key, ov::Dimension d) {
         self[key] = d.get_length();
     });
-    shape.def("__getitem__", [](const ov::Shape& v, size_t key) {
+    shape.def("__getitem__", [](const ov::Shape& v, int key) {
+        if (key < 0)
+            key += v.size();
         return v[key];
     });
 
+    shape.def("__getitem__", [](const ov::Shape& v, py::slice slice) {
+        size_t start, stop, step, slicelength;
+        if (!slice.compute(v.size(), &start, &stop, &step, &slicelength))
+            throw py::error_already_set();
+        ov::Shape result(slicelength);
+        for (size_t i = 0; i < slicelength; ++i) {
+            result[i] = v[start];
+            start += step;
+        }
+        return result;
```

Comment:
> I see that implementation is repeating for both classes.

It's almost entirely the same except for one thing: PartialShape doesn't have a constructor like Shape, so I have to use the `resize` method for the latter to adjust the size. Exporting the function in common.hpp would work only for Shape, while it wouldn't work for PartialShape. I'm trying to find a way to solve this problem.

Reply:
I managed to find a solution, it's not the most elegant one, but now I'm using the function for both. I simply create an instance of the class depending on which object I'm calculating (whether it's Shape or PartialShape), and then I pass it as a parameter to the get_slice function implemented inside common.hpp.

I say it's not elegant because it would be better to call
`return Common::shape_helper::get_slice(param...)` inside the `__getitem__` function. Unfortunately, I couldn't find a better method where I create only one function for both and call it as mentioned above. Another solution would be to create two separate functions, implementing the body inside common.hpp, first defining a function for `ov::Shape` and then the other for `ov::PartialShape`, but at this point instead of linking them so simply, it's better for me to keep them inside the `__getitem__` function.

Reply:
yep, it's ok for right now

Reply:
This can always be done after the merge if needed. For now let's resolve the conversation.